pkg/tlog/entry.go

```diff
@@ -88,14 +99,93 @@ func NewEntry(body []byte, integratedTime int64, logIndex int64, logID []byte, s
 	}
 
 	if inclusionProof != nil {
-		entry.logEntryAnon.Verification = &models.LogEntryAnonVerification{
-			InclusionProof: inclusionProof,
+		hashes := make([][]byte, len(inclusionProof.Hashes))
+		for i, s := range inclusionProof.Hashes {
+			hashes[i] = []byte(s)
```

Comment:
Do we need to `hex.DecodeString` here as well? https://github.com/sigstore/rekor/blob/c0f3b8c639cb52bb2fa760465ad7d49e2ff19570/pkg/api/entries.go#L96

Reply:
> Can we also see if we have test coverage for verifying proofs from both Rekor v1 and v2?

This is mostly covered by the e2e tests which exercise both v1 and v2. They didn't catch this bug because this is now a deprecated function. I added a new unit test which tests this path and catches this bug.